### Importación de datos



In [8]:
import pandas as pd

url = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_1%20.csv"
url2 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_2.csv"
url3 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_3.csv"
url4 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_4.csv"

tienda = pd.read_csv(url)
tienda2 = pd.read_csv(url2)
tienda3 = pd.read_csv(url3)
tienda4 = pd.read_csv(url4)

tienda.head()

Producto Categoría del Producto    Precio  Costo de envío  \
0  Asistente virtual           Electrónicos  164300.0          6900.0   
1    Mesa de comedor                Muebles  192300.0          8400.0   
2      Juego de mesa               Juguetes  209600.0         15900.0   
3         Microondas      Electrodomésticos  757500.0         41000.0   
4   Silla de oficina                Muebles  335200.0         20200.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      16/01/2021      Pedro Gomez          Bogotá             4   
1      18/05/2022  Beatriz Morales        Medellín             1   
2      15/03/2021   Juan Fernandez       Cartagena             1   
3      03/05/2022   Juan Fernandez            Cali             4   
4      07/11/2020    Maria Alfonso        Medellín             5   

       Método de pago  Cantidad de cuotas       lat       lon  
0  Tarjeta de crédito                   8   4.60971 -74.08175  
1  Tarjeta de crédito                   4   6.25184 -75.56359  
2  Tarjeta de crédito                   1  10.39972 -75.51444  
3               Nequi                   1   3.43722 -76.52250  
4               Nequi                   1   6.25184 -75.56359

#0. Análisis Exploratorio y fusion de dataframes



Voy a unir todo en un mismo dataframe y le voy a agregar una columna con el número de tienda

In [12]:
#Agrego columna tienda
tienda['tienda'] = 1
tienda2['tienda'] = 2
tienda3['tienda'] = 3
tienda4['tienda'] = 4

tienda.head()

Producto Categoría del Producto    Precio  Costo de envío  \
0  Asistente virtual           Electrónicos  164300.0          6900.0   
1    Mesa de comedor                Muebles  192300.0          8400.0   
2      Juego de mesa               Juguetes  209600.0         15900.0   
3         Microondas      Electrodomésticos  757500.0         41000.0   
4   Silla de oficina                Muebles  335200.0         20200.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      16/01/2021      Pedro Gomez          Bogotá             4   
1      18/05/2022  Beatriz Morales        Medellín             1   
2      15/03/2021   Juan Fernandez       Cartagena             1   
3      03/05/2022   Juan Fernandez            Cali             4   
4      07/11/2020    Maria Alfonso        Medellín             5   

       Método de pago  Cantidad de cuotas       lat       lon  tienda  
0  Tarjeta de crédito                   8   4.60971 -74.08175       1  
1  Tarjeta de crédito                   4   6.25184 -75.56359       1  
2  Tarjeta de crédito                   1  10.39972 -75.51444       1  
3               Nequi                   1   3.43722 -76.52250       1  
4               Nequi                   1   6.25184 -75.56359       1

In [14]:
#contateno las 4 tiendas en 1 solo dataframe
df=pd.concat([tienda,tienda2,tienda3,tienda4])

In [15]:
df.head()

Producto Categoría del Producto    Precio  Costo de envío  \
0  Asistente virtual           Electrónicos  164300.0          6900.0   
1    Mesa de comedor                Muebles  192300.0          8400.0   
2      Juego de mesa               Juguetes  209600.0         15900.0   
3         Microondas      Electrodomésticos  757500.0         41000.0   
4   Silla de oficina                Muebles  335200.0         20200.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      16/01/2021      Pedro Gomez          Bogotá             4   
1      18/05/2022  Beatriz Morales        Medellín             1   
2      15/03/2021   Juan Fernandez       Cartagena             1   
3      03/05/2022   Juan Fernandez            Cali             4   
4      07/11/2020    Maria Alfonso        Medellín             5   

       Método de pago  Cantidad de cuotas       lat       lon  tienda  
0  Tarjeta de crédito                   8   4.60971 -74.08175       1  
1  Tarjeta de crédito                   4   6.25184 -75.56359       1  
2  Tarjeta de crédito                   1  10.39972 -75.51444       1  
3               Nequi                   1   3.43722 -76.52250       1  
4               Nequi                   1   6.25184 -75.56359       1

In [17]:
df.tail()

Producto   Categoría del Producto    Precio  Costo de envío  \
2353     Mesa de comedor                  Muebles  227700.0         12600.0   
2354    Silla de oficina                  Muebles  342000.0         18600.0   
2355        Vaso térmico     Deportes y diversión   36700.0          2400.0   
2356        Set de ollas  Artículos para el hogar  151800.0          8300.0   
2357  Cuerda para saltar     Deportes y diversión   11200.0          1300.0   

     Fecha de Compra        Vendedor Lugar de Compra  Calificación  \
2353      12/02/2020   Camila Rivera        Medellín             5   
2354      17/05/2022   Felipe Santos     Bucaramanga             5   
2355      03/05/2022   Camila Rivera          Bogotá             5   
2356      15/06/2020  Blanca Ramirez            Cali             5   
2357      17/11/2021   Felipe Santos          Bogotá             5   

          Método de pago  Cantidad de cuotas      lat       lon  tienda  
2353  Tarjeta de crédito                   1  6.25184 -75.56359       4  
2354  Tarjeta de crédito                   3  7.12539 -73.11980       4  
2355  Tarjeta de crédito                   1  4.60971 -74.08175       4  
2356  Tarjeta de crédito                   1  3.43722 -76.52250       4  
2357  Tarjeta de crédito                   1  4.60971 -74.08175       4

In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 9435 entries, 0 to 2357
Data columns (total 13 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Producto                9435 non-null   object 
 1   Categoría del Producto  9435 non-null   object 
 2   Precio                  9435 non-null   float64
 3   Costo de envío          9435 non-null   float64
 4   Fecha de Compra         9435 non-null   object 
 5   Vendedor                9435 non-null   object 
 6   Lugar de Compra         9435 non-null   object 
 7   Calificación            9435 non-null   int64  
 8   Método de pago          9435 non-null   object 
 9   Cantidad de cuotas      9435 non-null   int64  
 10  lat                     9435 non-null   float64
 11  lon                     9435 non-null   float64
 12  tienda                  9435 non-null   int64  
dtypes: float64(4), int64(3), object(6)
memory usage: 1.3+ MB


Total 9435 ventas entre las 4 tiendas, no hay campos nulos

#1. Análisis de facturación

# 2. Ventas por categoría

# 3. Calificación promedio de la tienda


# 4. Productos más y menos vendidos

# 5. Envío promedio por tienda